### ISSUE TITLE
Header names are transformed into lowercase

### ISSUE THREAD

**Issue by FengWei-Plasmatic** [rangav/thunder-client-support#458]
**Describe the bug**
Uppercase header names are transformed into lowercase when sending an api request. This conflicts with AWS API Gateway since it checks header names in a case-sensitive way.

**To Reproduce**

Create a new request, enter valid url, and enter a header name with any uppercase characters, then send.
eg. Entering the header 'X-MY-CUSTOM-HEADER' will send it as 'x-my-custom-header'

**Expected behavior**
Header names retain their case.

**Platform:** 
 - OS: Windows 10
 - vscode version: 1.63.2
 - extension version: 1.11.1

**Comment by rogueturnip** [comment]
Just want to mention that https://www.rfc-editor.org/rfc/rfc7230 specifies
```
[3.2](https://www.rfc-editor.org/rfc/rfc7230#section-3.2).  Header Fields
   Each header field consists of a case-insensitive field name followed
   by a colon (":"), optional leading whitespace, the field value, and
   optional trailing whitespace.
```
AWS API Gateway is technically non-complient.  But the RFC doesn't say the client should lowercase the header it's sending.  I think the solution would be to send the headers as entered.  Not working with AWS API Gateway is a pretty big limitation.

**Comment by rangav** [comment]
Hi @FengWei-Plasmatic thanks for reporting the bug, will fix it asap.

**Comment by rangav** [comment]
Hi All, the library I am using is https://github.com/sindresorhus/got, which is transforming to all lowercase. Don't see any option to send raw headers

**Comment by rogueturnip** [comment]
No option to use axios?  A fork of got?  Pr to got that optionally allows case to be maintained?  Just thinking of some ideas.

**Comment by jsbravoc** [comment]
Perhaps this could work https://github.com/sindresorhus/got/blob/main/documentation/examples/uppercase-headers.js, using a custom `transformHeader` function that sets the headers to what was received on the inputs

**Comment by rangav** [comment]
Thanks @jsbravoc sharing the link, will look into it.

**Comment by rangav** [comment]
Hi all, this feature is now implemented, there will be new `vscode setting` to send headers in `same case`, this is implemented  using the custom hook.

You need to enable this setting, to send headers in **same case**, The setting is called `Headers Case Sensitive`

https://github.com/sindresorhus/got/blob/main/documentation/9-hooks.md

**Comment by rogueturnip** [comment]
This is great!  What version should we see this setting appear?  I was just looking at 1.12.1 and don't see it in there.

**Comment by rangav** [comment]
Sorry it will be in next update `v1.13.0`

**Comment by rangav** [comment]
This feature is now implemented in v1.12.2

see all features released
https://github.com/rangav/thunder-client-support/releases/tag/v1.12.2

Publishing another short release to fix 2 critical bugs #478, #480

Let me know if its working or not
